trackstop.lua

```diff
@@ -236,7 +260,170 @@ function RollerOverlay:init()
   }
 end
 
+selected_stop = selected_stop or nil
+
+ReorderStopsWindow = defclass(ReorderStopsWindow, widgets.Window)
+ReorderStopsWindow.ATTRS {
+  frame={t=4,l=60,w=45, h=28},
+  frame_title='Reorder Stops',
+  view_id='main',
+}
+
+local SELECT_STOP_HINT = 'Select a stop to move'
+local SELECT_ANOTHER_STOP_HINT = 'Select another stop on the same route'
+
+function ReorderStopsWindow:init()
+  self:addviews{
+    widgets.Label{
+      frame={t=0,l=0},
+      view_id='hint',
+      text=SELECT_STOP_HINT,
+    },
+    widgets.List{
+      view_id='routes',
+      frame={t=1,l=1},
+      choices={},
+      on_select=function(index, item)
+        if not item then return end
+        if item.type == 'stop' then
+          local item_pos = df.global.plotinfo.hauling.routes[item.route_index].stops[item.stop_index].pos
+          df.global.game.main_interface.recenter_indicator_m.x = item_pos.x
+          df.global.game.main_interface.recenter_indicator_m.y = item_pos.y
+          df.global.game.main_interface.recenter_indicator_m.z = item_pos.z
+          dfhack.gui.revealInDwarfmodeMap(item_pos, true)
+        end
+      end,
+      on_submit=function(index, item)
+        if selected_stop then
+          local hauling = df.global.plotinfo.hauling
+          local routes = hauling.routes
+          local view_stops = hauling.view_stops
+          local route = routes[item.route_index]
+
+          -- rearrange stops
+          if item.type == 'stop' then
+            local stop_index = item.stop_index
+
+            -- don't allow moving stops to a different route
```

Comment:
why not?

Reply:
Yeah this might be fine, I'll revisit it for sure.

Reply:
Now possible